### ISSUE TITLE
Problem editing quotas

### ISSUE THREAD

**Issue by bandicootdancing** [eerotal/LibreSignage#71]
Having an issue changing quotas. I found 00-default.php in the config/quota directory. I first tried 01-custom.php in that same directory and then saw ``Create a custom quota override file in conf/quota/ instead.`` The quota directory doesn't exist in the conf/ directory but once I make it the site becomes inaccessible. I tried creating 01-custom.php in there anyway but still could access the site. 

What is the correct way to increase quotas?

**Comment by eerotal** [comment]
The right directory to use is ``[INSTALL_DIR]/config/quota``. Make sure the new file you create is valid PHP and that it returns an array. In fact, you can just copy the default quota config 00-default.php to ``config/quota/01-custom.php`` and modify that.

If this doesn't work, there might be something wrong with the config loader code, although I don't remember having any problems with it before.

**Comment by bandicootdancing** [comment]
Probably just an error on my end then. Possibly the wrong place since I added the 00-custom.php to ``/var/www/[address]/config/quota``

**Comment by eerotal** [comment]
That does seem like the correct path though, but you can try it once more to see if it works.

**Comment by bandicootdancing** [comment]
Just a typo on my end in the last reply. I did ``cp 00-default.php 01-custom.php`` and no changes even after restarting Apache and logging out/in to LibreSignage.

**Comment by eerotal** [comment]
Do you mean the site was inaccessible or the changes did not work?

**Comment by bandicootdancing** [comment]
Changes did not work. The only time the site was inaccessible was when I tried creating the quota directory under ``/var/www/[address]/config/conf/quota/`` The only reason I tried there was because the 00-default.php mentions ``conf/quota/`` The quota I'm trying to change is the amount of slides. When logging in I still see 10 as the limit. I just tested to make sure and I'm limited to 10 slides with ``You can't create any more slides.``

```
<?php

/*
*  Default LibreSignage quota config. Don't edit this file directly.
*  Create a custom quota override file in conf/quota/ instead.
*/

return [
        'slides' => [
                'limit' => 20,
                'description' => 'Slides'
        ],
        'api_rate' => [
                'limit' => 200,
                'description' => 'API quota (calls/'.gtlim('API_RATE_T').'s)'
        ]
];
```

**Comment by eerotal** [comment]
Ah, I looked over the code a bit and there's a thing I have overlooked: changes to quotas are only applied to new users. The changes should take effect if you create a new user and login with that one. I'll need to fix this for v1.0.1.

**Comment by bandicootdancing** [comment]
Thanks, a new user did work with an increased quota. This is a fresh install so easy enough just to use a new user for now.

**Comment by eerotal** [comment]
Great that it works!

**Comment by infomaniac50** [comment]
This is more of a problem with storing settings in user accounts without a way to edit them. There are a number of configs in the user profile that are not exposed via the user account page.

These are a couple of settings I can remember off-hand.

- Maximum number slides
- API Quota

You can actually update the configs for each user if you edit the save files by hand. I believe the format is var_export or serialize. I not sure which but it is something PHP native. I would prefer JSON storage or maybe even SQLite.

**Comment by eerotal** [comment]
I wonder whether I should make quotas system-wide or leave them stored in the user data but make it possible to change quotas of individual users via the web interface.

**Comment by infomaniac50** [comment]
Is there a good use case to have separate quotas? My use case doesn't require quotas at all.

### Following the Project Goals

If we follow the project goals then I would have to say make quotas system-wide.

- **Create a system that's both easy to set up and easy to use**.
Right now the quota system is difficult. You have to edit the user profile or create a new user to change the quota. I think the expectation is that if you change the quota in the config then all the quotas will change.

- **Avoid scope creep**.
If there isn't a really good reason to have it then leave it out. Or at the very least set it aside for later. Software development is a continuous process. We can always add it back if enough users want it.

- **Keep it simple**.
It's complicated because user quotas are data-bound but the defaults are set in a static config. This also makes debugging hard because the profile may not match up with the defaults.

**Comment by JasperKuiper** [comment]
In general I'd prefer all settings to be data-bound, not static config. 

You could introduce a "super admin" user to set those current static configs for a future use case (e.g. Hosting party responsible for system performance) and "admin" for Customer user management.

A possible use case for setting user quota's:
Cafeteria department is awarded max 2 slides in a "cafeteria" queue used on different displays around the campus, where displays show multiple queues.

**Comment by eerotal** [comment]
I think I'll opt for the static config quotas at least for now. Originally I decided to implement quotas just so that there's some limits on what users can do. Quotas prevent users from filling up the whole server with data for example.

As @infomaniac50 mentioned, static config also follows the project goals a bit better, not to mention that implementing separate "super admin" and "admin" permissions would be a lot of work.

**Comment by eerotal** [comment]
And this issue is now fixed starting from the next patch release v1.0.1. As I mentioned in my earlier comment, quota limits aren't stored in userdata anymore, so any changes to quotas are applied to all users.
